clouddriver-ecs/src/main/java/com/netflix/spinnaker/clouddriver/ecs/deploy/ops/CreateServerGroupAtomicOperation.java

```diff
@@ -54,88 +59,188 @@ private static Task getTask() {
+    containerEnvironment.add(new KeyValuePair().withName("CLOUD_DETAIL").withValue(description.getFreeFormDetails()));
+
+    PortMapping portMapping = new PortMapping()
+      .withHostPort(0)
+      .withContainerPort(description.getContainerPort())
+      .withProtocol(description.getPortProtocol() != null ? description.getPortProtocol() : "tcp");
+
+    Collection<PortMapping> portMappings = new LinkedList<>();
+    portMappings.add(portMapping);
+
+    ContainerDefinition containerDefinition = new ContainerDefinition()
+      .withName(serverGroupVersion)
+      .withEnvironment(containerEnvironment)
+      .withPortMappings(portMappings)
+      .withCpu(description.getComputeUnits())
+      .withMemoryReservation(description.getReservedMemory())
+      .withImage(description.getDockerImageAddress());
+
+    Collection<ContainerDefinition> containerDefinitions = new LinkedList<>();
+    containerDefinitions.add(containerDefinition);
+
+    RegisterTaskDefinitionRequest request = new RegisterTaskDefinitionRequest()
+      .withContainerDefinitions(containerDefinitions)
+      .withFamily(getFamilyName());
+
+    updateTaskStatus("Creating Amazon ECS Task Definition...");
+    RegisterTaskDefinitionResult registerTaskDefinitionResult = ecs.registerTaskDefinition(request);
+    updateTaskStatus("Done creating Amazon ECS Task Definition...");
+
+    return registerTaskDefinitionResult.getTaskDefinition();
+  }
+
+  private Service createService(TaskDefinition taskDefinition) {
+    AmazonECS ecs = getAmazonEcsClient();
+    String serviceName = getNextServiceName();
+    Collection<LoadBalancer> loadBalancers = new LinkedList<>();
+    loadBalancers.add(getLoadBalancer());
+
+    Integer desiredCapacity = getDesiredCapacity();
+    String taskDefinitionArn = taskDefinition.getTaskDefinitionArn();
+
+    DeploymentConfiguration deploymentConfiguration = new DeploymentConfiguration()
+      .withMinimumHealthyPercent(50)
+      .withMaximumPercent(100);
+
+    CreateServiceRequest request = new CreateServiceRequest()
+      .withServiceName(serviceName)
+      .withDesiredCount(desiredCapacity != null ? desiredCapacity : 1)
+      .withCluster(description.getEcsClusterName())
+      .withRole(description.getIamRole())
+      .withLoadBalancers(loadBalancers)
+      .withTaskDefinition(taskDefinitionArn)
+      .withDeploymentConfiguration(deploymentConfiguration);
+
+    updateTaskStatus(String.format("Creating %s of %s with %s for %s.",
+      desiredCapacity, serviceName, taskDefinitionArn, description.getCredentialAccount()));
+
+    Service service = ecs.createService(request).getService();
+
+    updateTaskStatus(String.format("Done creating %s of %s with %s for %s.",
+      desiredCapacity, serviceName, taskDefinitionArn, description.getCredentialAccount()));
+
+    return service;
+  }
+
+  private void createAutoScalingGroup(Service service) {
```

Comment:
For debugging purposes it might be nice to return the `RegisterScalableTargetResult` from this method.  You could also log something useful from that `RegisterScalableTargetResult`

Reply:
Updated